### PR TITLE
[DinoMod] Dinosaurs Attack Bright Nights!

### PR DESCRIPTION
**Pull request by LyleSY**
#### Summary

SUMMARY: Mods "Catch up with DinoMod updates"

#### Purpose of change

Feature and content parity with DDA

#### Describe the solution

A port of the last six months or so of DDA DinoMod development. No DinoLabFinale sadly, lab templates and/or furniture work differently between BN and DDA. That means no friendly bionic dino mounts for now as mid-late game lab rewards.  Otherwise there is good feature and content parity here.

#### Describe alternatives you've considered

Only some of the dinos. A total DinoLabFinale rework for BN compatibility.

#### Testing

Just a JSON port

#### Additional context

Everything is better with dinosaurs

### CODE REVIEW

**Comment by Kenan2000** [comment]
Sci-fi dinos!Grr!

File: data/mods/DinoMod/monsters/dinosaur.json [hunk 1]
```diff
@@ -0,0 +1,1234 @@
...
+    "dodge": 1,
+    "armor_bash": 4,
+    "armor_cut": 1,
+    "hp": 150,
+    "death_function": [ "NORMAL" ],
+    "zombify_into": "mon_zilophosaurus",
+    "description": "A medium dinosaur with sharp teeth and two prominent bony crests on its head.",
+    "reproduction": { "baby_egg": "egg_dilophosaurus", "baby_count": 3, "baby_timer": 18 },
+    "baby_flags": [ "SPRING", "SUMMER" ],
+    "biosignature": { "biosig_item": "feces_bird", "biosig_timer": 3 },
```
Comment (Coolthulhu) [inline data/mods/DinoMod/monsters/dinosaur.json:73]: Not supported

Comment (LyleSY) [reply data/mods/DinoMod/monsters/dinosaur.json:73]: Which line(s)?

Comment (Coolthulhu) [reply data/mods/DinoMod/monsters/dinosaur.json:73]: Biosignature

**Comment by Coolthulhu** [comment]
Attempted to run and got the following errors:
```
src/generic_factory.h:404 [int_id<T> generic_factory<T>::convert(const string_id<T>&, const int_id<T>&) const [with T = furn_t]] invalid furniture id "f_machinery_old"
01:34:38.619 ERROR : src/generic_factory.h:404 [int_id<T> generic_factory<T>::convert(const string_id<T>&, const int_id<T>&) const [with T = furn_t]] invalid furniture id "f_console_broken"
01:34:38.622 ERROR : src/generic_factory.h:404 [int_id<T> generic_factory<T>::convert(const string_id<T>&, const int_id<T>&) const [with T = furn_t]] invalid furniture id "f_centrifuge"
01:34:46.428 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in broth_ingredients is not a valid item template
01:34:46.428 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in eggs_bird is not a valid item template
01:34:46.428 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in fish_soup_ingredients is not a valid item template
01:34:46.430 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in inline_recipe_boiled_egg is not a valid item template
01:34:46.430 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in inline_recipe_brioche is not a valid item template
01:34:46.430 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in inline_recipe_brownie is not a valid item template
01:34:46.430 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in inline_recipe_brownie_weed is not a valid item template
01:34:46.430 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in inline_recipe_cake2 is not a valid item template
01:34:46.431 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in inline_recipe_choc_pancakes is not a valid item template
01:34:46.431 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in inline_recipe_choc_waffles is not a valid item template
01:34:46.431 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in inline_recipe_deluxe_beans is not a valid item template
01:34:46.431 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in inline_recipe_deluxe_beansnrice is not a valid item template
01:34:46.431 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in inline_recipe_deluxe_eggs is not a valid item template
01:34:46.431 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in inline_recipe_deluxe_rice is not a valid item template
01:34:46.431 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in inline_recipe_eggnog is not a valid item template
01:34:46.431 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in inline_recipe_eggnog_spiked is not a valid item template
01:34:46.432 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in inline_recipe_frenchtoast is not a valid item template
01:34:46.432 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in inline_recipe_fried_brain is not a valid item template
01:34:46.432 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in inline_recipe_fruit_pancakes is not a valid item template
01:34:46.432 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in inline_recipe_fruit_waffles is not a valid item template
01:34:46.432 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in inline_recipe_icecream_custard is not a valid item template
01:34:46.432 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in inline_recipe_leverpostej is not a valid item template
01:34:46.433 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in inline_recipe_mayonnaise is not a valid item template
01:34:46.433 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in inline_recipe_mutagen_bird is not a valid item template
01:34:46.433 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in inline_recipe_pancakes is not a valid item template
01:34:46.433 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in inline_recipe_powder_eggs is not a valid item template
01:34:46.433 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in inline_recipe_scrambled_eggs is not a valid item template
01:34:46.433 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in inline_recipe_scrambledeggsandbrain is not a valid item template
01:34:46.434 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in inline_recipe_space_cake is not a valid item template
01:34:46.434 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in inline_recipe_sponge_cake is not a valid item template
01:34:46.434 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in inline_recipe_stewed_tripe is not a valid item template
01:34:46.434 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in inline_recipe_waffles is not a valid item template
01:34:46.435 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in meat_soup_ingredients is not a valid item template
01:34:46.435 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in veggy_soup_ingredients is not a valid item template
01:34:46.435 ERROR : src/requirements.cpp:423 [void component::check_consistency(const string&) const] egg_bird_unfert in woods_soup_ingredients is not a valid item template
```

**Comment by LyleSY** [comment]
It looks like the vanilla labfinale map that DinoLabFinale is based on has changed a lot between DDA and BN and would need a total rework for compatibility.

**Comment by LyleSY** [comment]
The Travis test failure doesn't look related, it's about reading books

**Comment by Coolthulhu** [comment]
That test keeps failing randomly, I'll need to do something about it.

Did you test the mod manually? I'm going to run automatic testing and some manual testing, but I might not catch all errors.

**Comment by Coolthulhu** [comment]
```
src/generic_factory.h:404 [int_id<T> generic_factory<T>::convert(const string_id<T>&, const int_id<T>&) const [with T = furn_t]] invalid furniture id "f_machinery_old"
14:13:02.123 ERROR : src/generic_factory.h:404 [int_id<T> generic_factory<T>::convert(const string_id<T>&, const int_id<T>&) const [with T = furn_t]] invalid furniture id "f_console_broken"
14:13:02.125 ERROR : src/generic_factory.h:404 [int_id<T> generic_factory<T>::convert(const string_id<T>&, const int_id<T>&) const [with T = furn_t]] invalid furniture id "f_centrifuge"
```

All of those are terrain types, but used as furniture.
For it to work, the type needs to be changed from furniture to terrain and the id from `f_*` to `t_*`.

There are also some warnings about waffles and pancakes having too high average calories because dino eggs push average calories up, but those warnings aren't very important.

**Pull request merged**